src/plugins/core/header_visibility.ts

```diff
@@ -97,7 +97,11 @@ export class HeaderVisibilityPlugin extends CorePlugin {
     return;
   }
 
-  canRemoveHeaders(sheetId: UID, dimension: Dimension, elements: HeaderIndex[]): boolean {
+  checkElementsIncludeAllVisibleHeaders(
+    sheetId: UID,
+    dimension: Dimension,
+    elements: HeaderIndex[]
+  ): boolean {
     const visibleHeaders = this.getAllVisibleHeaders(sheetId, dimension);
     return !includesAll(elements, visibleHeaders);
```

Comment:
```suggestion
    return includesAll(elements, visibleHeaders);
```
remove the `!` and update the calls of this method. With the "!" it does the opposite of wha its name says